watertap/unit_models/reverse_osmosis_0D.py

```diff
@@ -890,154 +853,75 @@ def calculate_scaling_factors(self):
+                sf = iscale.get_scaling_factor(self.Kf_io[ind]) * \
+                        iscale.get_scaling_factor(self.dh)
                 iscale.constraint_scaling_transform(c, sf)
 
         if hasattr(self, 'eq_N_Re_io'):
-            for ind, c in self.eq_N_Re_io.items():
-                sf = iscale.get_scaling_factor(self.N_Re_io[ind])
-                iscale.constraint_scaling_transform(c, sf*1e4)
-
-        if hasattr(self, 'eq_N_Sc_io'):
-            for ind, c in self.eq_N_Sc_io.items():
-                sf = iscale.get_scaling_factor(self.N_Sc_io[ind])
-                iscale.constraint_scaling_transform(c, sf*1e3)
-
-        if hasattr(self, 'eq_N_Sh_io'):
-            for ind, c in self.eq_N_Sh_io.items():
-                sf = iscale.get_scaling_factor(self.N_Sh_io[ind])
+            for (t,x), c in self.eq_N_Re_io.items():
+                if x == 'in':
+                    prop_io = self.feed_side.properties_in[t]
+                elif x == 'out':
+                    prop_io = self.feed_side.properties_out[t]
+                sf = min(iscale.get_scaling_factor(v) for v in prop_io.flow_mass_phase_comp['Liq',:])
+                sf *= iscale.get_scaling_factor(self.dh)
                 iscale.constraint_scaling_transform(c, sf)
 
-        if hasattr(self, 'eq_area'):
-            sf = iscale.get_scaling_factor(self.area)
-            iscale.constraint_scaling_transform(self.eq_area, sf)
-
-        if hasattr(self, 'eq_pressure_change'):
-            for ind, c in self.eq_pressure_change.items():
-                sf = iscale.get_scaling_factor(self.deltaP[ind])
+        if hasattr(self, 'eq_N_Sc_io'):
+            for (t,x), c in self.eq_N_Sc_io.items():
+                if x == 'in':
+                    prop_io = self.feed_side.properties_in[t]
+                elif x == 'out':
+                    prop_io = self.feed_side.properties_out[t]
+                sf = iscale.get_scaling_factor(prop_io.visc_d_phase['Liq'])
                 iscale.constraint_scaling_transform(c, sf)
 
         if hasattr(self, 'eq_velocity_io'):
-            for ind, c in self.eq_velocity_io.items():
-                sf = iscale.get_scaling_factor(self.velocity_io[ind])
-                iscale.constraint_scaling_transform(c, sf*100.)
-
-        if hasattr(self, 'eq_friction_factor_darcy_io'):
-            for ind, c in self.eq_friction_factor_darcy_io.items():
-                sf = iscale.get_scaling_factor(self.friction_factor_darcy_io[ind])
-                iscale.constraint_scaling_transform(c, sf/10.)
+            for (t, x), c in self.eq_velocity_io.items():
+                if x == 'in':
+                    prop_io = self.feed_side.properties_in[t]
+                elif x == 'out':
+                    prop_io = self.feed_side.properties_out[t]
+                sf = iscale.get_scaling_factor(prop_io.flow_vol_phase['Liq'])
+                iscale.constraint_scaling_transform(c, sf)
```

Comment:
I was under the impression that with the new solver, we generally wouldn't need to apply `constraint_scaling_transform`. Was the inclusion necessary, or did you keep the transforms in since they were already there?

Reply:
So for the time being it seems necessary. When I eliminate the calls to `constraint_scaling_transform` from the RO models I get two test failures in the full_treatment_train folder:
```
========================================================================= short test summary info =========================================================================
FAILED watertap/examples/flowsheets/full_treatment_train/analysis/tests/test_flowsheets.py::test_flowsheet_two_stage - RuntimeError: The solver failed to converge to an...
FAILED watertap/examples/flowsheets/full_treatment_train/analysis/tests/test_flowsheets.py::test_flowsheet_1DRO - RuntimeError: The solver failed to converge to an opti...
FAILED watertap/property_models/tests/test_NaCl_prop_pack.py::TestNaClProperty_idaes::test_initialize_failure - Failed: DID NOT RAISE <class 'idaes.core.util.exceptions...
FAILED watertap/property_models/tests/test_seawater_prop_pack.py::TestSeawaterProperty_idaes::test_initialize_failure - Failed: DID NOT RAISE <class 'idaes.core.util.ex...
========================================================== 4 failed, 531 passed, 8 skipped in 236.68s (0:03:56) ===========================================================
```

Interestingly, as an experiment I merged #285 into this branch locally, I did not see these failures (and I think the four I see are not of much concern):
```
========================================================================= short test summary info =========================================================================
FAILED watertap/examples/flowsheets/lsrro/tests/test_lsrro.py::TestLSRRO_2Stage::test_optimize - assert 0.00045104115267494333 == 0.000451048 ± 4.5e-09
FAILED watertap/property_models/tests/test_NaCl_prop_pack.py::TestNaClProperty_idaes::test_initialize_failure - Failed: DID NOT RAISE <class 'idaes.core.util.exceptions...
FAILED watertap/property_models/tests/test_seawater_prop_pack.py::TestSeawaterProperty_idaes::test_initialize_failure - Failed: DID NOT RAISE <class 'idaes.core.util.ex...
FAILED watertap/unit_models/zero_order/tests/test_nanofiltration_zo.py::TestNFZO_w_default_removal::test_report - AssertionError: assert '\n============================...
========================================================== 4 failed, 531 passed, 8 skipped in 219.25s (0:03:39) ===========================================================
```

Note that we still may need to transform *some* constraints, though ideally those can be kept to the property packages. The new solver should make this the exception rather than the rule, however. I also don't think I spent much time finding a minimal set of constraints that needed transforming on the RO side.